contrib/Dockerfile.test

```diff
@@ -111,13 +111,14 @@ CMD ["script/critest.sh", "/tmp"]
 # Requires Rootless Docker/Podman/nerdctl with cgroup v2 delegation: https://rootlesscontaine.rs/getting-started/common/cgroup2/
 # (Rootless Docker/Podman/nerdctl prepares the UserNS, so we do not need to create UserNS by ourselves)
 FROM critest AS cri-in-userns
+RUN apt-get update && apt-get install -y systemd
 COPY contrib/Dockerfile.test.d/cri-in-userns/etc_containerd_config.toml /etc/containerd/config.toml
 COPY contrib/Dockerfile.test.d/cri-in-userns/docker-entrypoint.sh       /docker-entrypoint.sh
-ENTRYPOINT ["/docker-entrypoint.sh"]
+COPY contrib/Dockerfile.test.d/cri-in-userns/critest.sh /critest.sh
 # Skip "runtime should support unsafe sysctls": `container init caused:  write sysctl key fs.mqueue.msg_max: open /proc/sys/fs/mqueue/msg_max: permission denied`
 # Skip "runtime should support safe sysctls": `container init caused: write sysctl key kernel.shm_rmid_forced: open /proc/sys/kernel/shm_rmid_forced: permission denied`
 # Skip "should allow privilege escalation when (NoNewPrivis is) false": expected log "Effective uid: 0\n" (stream="stdout") not found in logs [{timestamp:{wall:974487519 ext:63761339984 loc:<nil>} stream:stdout log:Effective uid: 1000) }]
-CMD ["critest", "--ginkgo.skip=should support unsafe sysctls|should support safe sysctls|should allow privilege escalation when false"]
+ENTRYPOINT ["/critest.sh"]
```

Comment:
?

Reply:
Because I want to start systemd in podman, I modify the startup command. but this creates a new problem. systemd will cause the container to fail to exit normally.
$ cat critest.sh
```
#!/usr/bin/env bash

cat > /etc/systemd/system/critest.service << EOF
[Unit]
Description=critest script
[Service]
Type=simple
ExecStart=/bin/bash /docker-entrypoint.sh
ExecStop=/bin/sh -c 'sleep 10 && /bin/systemctl poweroff'
StandardOutput=/dev/stdout
StandardError=/dev/stderr
[Install]
WantedBy=default.target
EOF
systemctl enable critest.service
# shellcheck disable=SC2093
journalctl -f &
exec /lib/systemd/systemd; journalctl -fu critest.service
```

Reply:
systemd has been already executed via `ENTRYPOINT ["/docker-entrypoint.sh"]`

Reply:
`"Enabling cgroup v2 nesting"` has to occur before running systemd

Reply:
See also https://github.com/moby/moby/blob/master/hack/dind-systemd

Reply:
https://github.com/containerd/containerd/actions/runs/7053834109/job/19201721551?pr=9350

Through CI, we found that the current script has taken effect and all test cases are running normally. However, because the container is stopped using systemctl poweroff, the container exit code is 130.

<img width="1062" alt="image" src="https://github.com/containerd/containerd/assets/15009201/7aff5b4d-2de9-47c9-88b2-11f84bafc8a4">